internal/dynamicwatcher/dynamic_watcher_test.go

```diff
@@ -0,0 +1,138 @@
+package dynamicwatcher
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/package-operator/package-operator/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	toolscache "k8s.io/client-go/tools/cache"
+)
+
+type RestMapperMock struct {
```

Comment:
Maybe we should decide if want to define mocks before or after the tests, just to make everything a little more consistent